src/components/VideoPreview.tsx

```diff
@@ -1,33 +1,49 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useVideo } from '@/api/hooks'
 import routes from '@/config/routes'
-import VideoPreviewBase, { VideoPreviewBaseMetaProps } from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import VideoPreviewBase, {
+  VideoPreviewBaseMetaProps,
+  VideoPreviewPublisherProps,
+} from '@/shared/components/VideoPreviewBase/VideoPreviewBase'
+import { useDrafts } from '@/hooks'
 
-type VideoPreviewProps = {
+export type VideoPreviewProps = {
   id?: string
   progress?: number
+  isLoading?: boolean
   className?: string
-} & VideoPreviewBaseMetaProps
+} & VideoPreviewBaseMetaProps &
+  VideoPreviewPublisherProps
 
-const VideoPreview: React.FC<VideoPreviewProps> = ({ id, progress, className, ...metaProps }) => {
+const VideoPreview: React.FC<VideoPreviewProps> = ({ id, className, isLoading = false, ...metaProps }) => {
```

Comment:
To continue on the fact that we probably shouldn't use drafts outside of publisher screens - we probably should create a separate component for publishing video preview. Then we don't need to include all the additional logic on the consumer side